e2e/tests/regression/test_scheduling.robot

```diff
@@ -161,19 +161,108 @@ Test Replica Auto Balance Node Least Effort
     And Check volume 0 data is intact
 
 Test Data Locality
+    [Documentation]    Test that Longhorn builds a local replica on the engine node
     Given Create single replica volume 0 with replica on node 0    dataLocality=disabled    dataEngine=${DATA_ENGINE}
     When Attach volume 0 to node 1
     And Write data to volume 0
-    Then Volume 0 should have 0 replicas running on node 1
+    Then Volume 0 should have 0 running replicas on node 1
 
     When Update volume 0 data locality to best-effort
     Then Wait until volume 0 replica rebuilding started on node 1
-    And Volume 0 should have 1 replicas running on node 1
-    And Volume 0 should have 0 replicas running on node 0
+    And Volume 0 should have 1 running replicas on node 1
+    And Volume 0 should have 0 running replicas on node 0
 
     When Detach volume 0 from node 1
     And Wait for volume 0 detached
     And Attach volume 0 to node 2
     Then Wait until volume 0 replica rebuilding started on node 2
-    And Volume 0 should have 1 replicas running on node 2
-    And Volume 0 should have 0 replicas running on node 1
+    And Volume 0 should have 1 running replicas on node 2
+    And Volume 0 should have 0 running replicas on node 1
+
+Test Replica Deleting Priority With Best-effort Data Locality
+    [Documentation]    Test that Longhorn prioritizes deleting replicas on the same node
+    Given Set node 0 tags    AVAIL
+    And Set node 1 tags    AVAIL
+
+    ${avail_node_selector}=    Create List    AVAIL
+    When Create volume 0    dataLocality=best-effort    nodeSelector=${avail_node_selector}    dataEngine=${DATA_ENGINE}
+    And Attach volume 0 to node 2
+    And Wait for volume 0 degraded
+    Then Volume 0 should have running replicas on node 0
+    And Volume 0 should have running replicas on node 1
+    And Volume 0 should have 0 running replicas on node 2
+    And Check volume 0 works
+
+    When Update volume 0 replica count to 2
+    And Wait for volume 0 healthy
+    # Longhorn will prioritize deleting replicas on the same node to maintain the balance
+    # the replica on the node with more replicas than the others will be deleted
+    Then Volume 0 should have running replicas on node 0
+    And Volume 0 should have running replicas on node 1
+
```

Comment:
Hi @yangchiu 
Do we need to check if Volume 0 has 1 running replica on both `Node 0` and `Node 1`  in Step 9 of Case 2?
https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/test_scheduling.py#L1345